### ISSUE TITLE
Quantizing model appears to slow inference speed

### ISSUE THREAD

**Issue by itberrios** [Deci-AI/super-gradients#975]
### Describe the bug
When trying post-training quantization as described [here](https://docs.deci.ai/super-gradients/documentation/source/ptq_qat.html#post-training-quantization) with a pretrained YOLO-NAS S, the inference speed drops from ~15fps to ~4fps. I'm assuming that I'm doing something wrong when quantizing the model, just not sure what it is.

### To Reproduce
0. Install appropriate libraries
```
! pip install torch==1.11.0+cu113 torchvision==0.12.0+cu113 torchaudio==0.11.0 --extra-index-url https://download.pytorch.org/whl/cu113 &> /dev/null
! pip install super-gradients
! pip install pytorch-quantization==2.1.2 --extra-index-url https://pypi.ngc.nvidia.com &> /dev/null
! pip install matplotlib==3.1.3 &> /dev/null
! pip install --upgrade psutil==5.9.2 &> /dev/null
! pip install --upgrade pillow==7.1.2 &> /dev/null
```
1. Import pre-trained YOLO-NAS-s
```
from super_gradients.training import models
from super_gradients.common.object_names import Models

model = models.get("yolo_nas_s", pretrained_weights="coco").cuda()
model.eval();
```
2. Quantize model
```
from super_gradients.training.utils.quantization.selective_quantization_utils import SelectiveQuantizer

q_util = SelectiveQuantizer(
    default_quant_modules_calibrator_weights="max",
    default_quant_modules_calibrator_inputs="histogram",
    default_per_channel_quant_weights=True,
    default_learn_amax=False,
    verbose=True,
)
q_util.quantize_module(model)
model.eval();
```
3. Compare inference speed
    (DO this once with the regular model and again with the quantized model)
```
base_test_times = []
for i in range(100):
  # start clock and rpedict
  tic = time.perf_counter()
  preds = model.predict(image_in, iou=0.25, conf=0.30)

  # get xyxyc for input to object tracker
  img_preds = list(preds._images_prediction_lst)[0]
  dets = np.hstack((img_preds.prediction.bboxes_xyxy, 
                    np.c_[img_preds.prediction.confidence]))
  base_test_times.append(time.perf_counter() - tic)

print(f"Average Prediction time: {np.mean(base_test_times)} "
      f"Average FPS: {1/np.mean(base_test_times)}")
```

### Expected behavior
I would expect the quantized model to perform faster inference than the regular model.

**Comment by spsancti** [comment]
Hi!
You are running inference in Pytorch, so that's expected behavior. 
Pytorch side uses fake quantization that is still performed in FP32, so it adds computation time for scaling, rounding, etc. Basically, you only perform QAT on a quantized model in Pytorch, and then convert it for inference.

For faster inference, you will need an inference framework (only TensorRT is officially supported for quantized models).

Please refer to the next section of our [documentation](https://docs.deci.ai/super-gradients/documentation/source/ptq_qat.html#converting-quantized-models-to-onnx-for-inference) for more info.

**Comment by itberrios** [comment]
Thanks! This really helps clear things up!
